### PR TITLE
Send Mock Time Slot Notification Email

### PR DESCRIPTION
**Pull request by tdkent**
#### Note: Ethereal username and password environment variables are required to run this branch

After a request form is received and added to the database, a mock email is created with information for the customer about the requested time slot. The mock email is created using Ethereal. Ethereal emails expire after a few hours.

- The Appointment model has been expanded with fields for the current "Request Received" and future "Appointment Confirmation" public URLs. 
- The email is created after the form data is successfully added to the database. If email generation is successful, the public URL is added to the newly created Appointment document.
- The user is directed to the "Appointment Success" view, which has been updated with links to the mock email hosted by Ethereal and the appointment details view.
- The mock email link is also added to the Appointment Details view

- Adds `nodemailer` package to server
- Adds Ethereal username and password env variables

UPDATED 11/30: 

- updates to `emailHelper` (now `sendEmail`) helper
- `getSingleAppointment`: sort by date (desc) and limit to 1 result (_temp fix_)

### CODE REVIEW

File: server/src/controllers/appointments.controller.js [hunk 1]
```diff
@@ -1,29 +1,50 @@
...
-    // send success response
+    const newAppt = await appt.save();
+
+    // send mock appt request email, add url to the database
+    const emailPreviewUrl = await emailHelper({
+      email,
+      subject: "Appointment Request Received",
+      text: apptRequestConfirmationText(req.body),
+      html: apptRequestConfirmationHtml(req.body),
+    });
```
Comment (DalySoftware) [inline server/src/controllers/appointments.controller.js:39]: The signature here feels a little odd. emailHelper.js exposes these 2 methods to get the text and html, but then we only use those as for parameters into the other function it exposes.

```suggestion
    const emailPreviewUrl = await emailHelper({
      email,
      subject: "Appointment Request Received",
      formData: req.body
    });
```
If you had something like this then it seems better encapsulated and more cohesive? `emailHelper` is also not very descriptive, would `sendEmail` be an appropriate name?

File: server/src/utils/emailHelper.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          We’ve received your request for an appointment between ${timeRange.early} and ${timeRange.late} at ${formData.address}. Please note the date and time have not yet been finalized. We’ll confirm these details and notify you the night before your visit.
+        </p>
+        <p>If you have any questions or need assistance, feel free to contact us.</p><p>We look forward to seeing you soon!</p>
+        <p>Best regards,<br>RayVolution</p>
+      </div>
+    </div>
+  `;
+};
+
+const emailHelper = async ({ email, subject, text, html }) => {
```
Comment (DalySoftware) [inline server/src/utils/emailHelper.js:43]:
```suggestion
const emailHelper = async ({ toAddress, subject, text, html }) => {
```
`email` becomes a very overloaded term in this context, I assume this is meant to represent the `toAddress`

**Review by DalySoftware — COMMENTED** [review]
Page looks really nice 👍 

Is this a good time to introduce a proper `fetchSingleAppointment` which takes an appointment identifier? This branch breaks the site for someone with existing appointments if you get unlucky (as happened when I tested it).

File: server/src/controllers/appointments.controller.js (hunk 1 above)
Comment (tdkent) [reply server/src/controllers/appointments.controller.js:39]: Yeah I was struggling with the signature, but can't come up with a better way while still keeping `sendEmail` generic, or moving the text and email generation into the controller function itself

**Comment by tdkent** [comment]
> Page looks really nice 👍
> 
> Is this a good time to introduce a proper `fetchSingleAppointment` which takes an appointment identifier? This branch breaks the site for someone with existing appointments if you get unlucky (as happened when I tested it).

I made the branch assuming each user would have only appointment, but that doesn't really work since a single user could have multiple addresses they want visited, or in some cases multiple appointments to the same address. This aspect definitely needs to be figured out.

**Review by DalySoftware — APPROVED** [review]

**Pull request merged**